test/src/worker/browser-env/runner/test-runner/index.ts

```diff
@@ -427,6 +441,252 @@ describe("worker/browser-env/runner/test-runner", () => {
             });
         });
 
+        describe(`"${BrowserEventNames.runExpectMatcher}" event`, () => {
+            let browser: Browser;
+
+            beforeEach(() => {
+                global.expect = sandbox.stub() as unknown as ExpectWebdriverIO.Expect;
+
+                browser = mkBrowser_();
+                (BrowserAgent.prototype.getBrowser as SinonStub).resolves(browser);
+            });
+
+            afterEach(() => {
+                delete (global as Partial<{ expect: ExpectWebdriverIO.Expect }>).expect;
+            });
+
+            describe("should return error if", () => {
+                it("expect module is not found", done => {
+                    global.expect = undefined as unknown as ExpectWebdriverIO.Expect;
+
+                    const socket = mkSocket_() as BrowserViteSocket;
+                    socketClientStub.returns(socket);
+
+                    runWithEmitBrowserInit(socket).then(() => {
+                        socket.emit(
```

Comment:
I would make some helper function like:

```ts
const emitAsync = (socket, ...args) => new Promise(resolve => {
    socket.emit(...args, resolve);
}):
```

Then we could use it to write usual async tests, which look a lot simpler:

```ts
it("expect module is not found", async () => {
    ...
    await runWithEmitBrowserInit(socket);

    await emitAsync(socket, BrowserEventNames.runExpectMatcher, { name: "foo", args: [], scope: {} as MatcherState });

    assert.deepEqual(response, [{ pass: false, message: "Couldn't find expect module" }]);
    ....
});
```